### ISSUE TITLE
The `plutus-pab-examples` executable consumes a lot of memory.

### ISSUE THREAD

**Issue by VladimirLogachev** [input-output-hk/plutus-apps#420]
### Summary

We have set up a server just to be able to follow [the test-node instructions](https://github.com/input-output-hk/plutus-apps/tree/main/plutus-pab/test-node):

x86_64
12 CPU cores
24 GB RAM
80 GB swap

But the `plutus-pab-examples` executable seems to be killed due to OOM.

At block `11 752` the amount of swap used was 22 GB
At block `15 427` (slot `6 959 416`) the amount of swap used was 72 GB
And that's far from completion:
According to [the testnet explorer](https://explorer.cardano-testnet.iohkdev.io/en.html), at the moment of submitting the issue, the latest slot on testnet is `55 601 949`, block `3 474 971`

If it will help to find the problem, we can provide SSH access to the server.

<img width="1672" alt="Screenshot 2022-04-15 at 01 10 09" src="https://user-images.githubusercontent.com/17773003/163479969-5aeb95df-9e56-42e7-b510-4bb5410cb7d2.png">

### Steps to reproduce the behavior

https://github.com/input-output-hk/plutus-apps/tree/main/plutus-pab/test-node

### Actual Result

The pab examples sync process does not complete due to OOM

### Expected Result

The pab examples sync process completes

### System info

OS: Ubuntu
Version: 18.04
plutus-apps: [v2022-04-06](https://github.com/input-output-hk/plutus-apps/releases/tag/v2022-04-06)

**Comment by ak3n** [comment]
Could you please share your observations if there are any? when the memory usage increases? after the integration test contract endpoint activation?

**Comment by VladimirLogachev** [comment]
Integration test not started.
Please tell me your email/telegram/other contact, so that I could give you an access to the server.
It doesn't contain anything private at the moment.

**Comment by VladimirLogachev** [comment]
I've just tried running `plutus-pab-examples` with `--rollback-history 1000`, But I have no idea whether `1000` is a sane value or not.

It is still synchronizing, `Current block: 26 939. Current slot: 8 253 200`, which is more successful, and memory consumption looks more healthy.

<img width="1625" alt="Screenshot 2022-04-19 at 19 34 33" src="https://user-images.githubusercontent.com/17773003/164041380-26859d47-efdb-403a-a876-2f0639d22a66.png">

**Comment by ak3n** [comment]
Yeah, reducing the rollback history might help a bit but is not a general case. Working on the fix.

**Comment by waalge** [comment]
If it helps anyone: setting the dev options to the current slot, cleaning the contract db (`rm my.db` + `migrate`), and rerunning leads to a PAB service that is stable for a while. Enough for developing with at least.

**Comment by agocorona** [comment]
> If it helps anyone: setting the dev options to the current slot, cleaning the contract db (`rm my.db` + `migrate`), and rerunning leads to a PAB service that is stable for a while. Enough for developing with at least.

To make it more explicit,  unconment the lines [here](https://github.com/input-output-hk/plutus-apps/blob/fdb0eec477fc0f99f5edfdbc11c041dbd35ca6bb/plutus-pab/test-node/testnet/pab-config.yml#L51-L56) of pab-config.yml  to:

developmentOptions:
  pabRollbackHistory: null
  pabResumeFrom:
    #{ "tag" : "PointAtGenesis" }
    { "tag" : "Point"
    , "pointBlockId" : " enter here the last block id from the log of the cardano node"
    , "pointSlot" : { "getSlot" : the slot }
    }

And now it seems to synchronize well. Following https://youtu.be/X6AyZIZ0vaE?t=769

**Comment by agocorona** [comment]
I would like to know why the PAB app needs to sync
It may be important for the knowledge of the cardano blockchain.
Is it because of the possibility of deep pullbacks?

**Comment by VladimirLogachev** [comment]
up
